core/domain/email_manager.py

```diff
@@ -164,5 +178,71 @@ def send_post_signup_email(user_id):
         EMAIL_FOOTER.value)
 
     _send_email(
-        user_id, feconf.SYSTEM_COMMITTER_ID, email_models.INTENT_SIGNUP,
+        user_id, feconf.SYSTEM_COMMITTER_ID, feconf.EMAIL_INTENT_SIGNUP,
         email_subject, email_body)
+
+
+def require_valid_intent(intent):
+    if intent not in feconf.VALID_MODERATOR_ACTIONS:
+        raise Exception('Unrecognized email intent: %s' % intent)
+
+
+def _get_email_config(intent):
+    require_valid_intent(intent)
+    return config_domain.Registry.get_config_property(
+        feconf.VALID_MODERATOR_ACTIONS[intent]['email_config'])
+
+
+def get_draft_moderator_action_email(intent):
+    """Returns a draft of the text of the body for an email sent immediately
+    following a moderator action.
+
+    If feconf.REQUIRE_EMAIL_ON_MODERATOR_ACTION is set to False, returns None.
+    """
+    if not feconf.REQUIRE_EMAIL_ON_MODERATOR_ACTION:
+        return None
+
+    require_moderator_email_prereqs_are_satisfied(intent)
+    return _get_email_config(intent).value
+
+
+def require_moderator_email_prereqs_are_satisfied(intent):
+    """Raises an exception if, for any reason, moderator emails cannot be sent.
+    """
+    if not feconf.REQUIRE_EMAIL_ON_MODERATOR_ACTION:
+        raise Exception(
+            'For moderator emails to be sent, please ensure that '
+            'REQUIRE_EMAIL_ON_MODERATOR_ACTION is set to True.')
+    if not feconf.CAN_SEND_EMAILS_TO_USERS:
+        raise Exception(
+            'For moderator emails to be sent, please ensure that '
+            'CAN_SEND_EMAILS_TO_USERS is set to True.')
+
+    email_config = _get_email_config(intent)
+    if email_config.value == email_config.default_value:
+        raise Exception(
+            'Please ensure that the value for the admin config property '
+            '%s is set, before allowing moderator emails for %s to be '
+            'sent.' % (intent, email_config.name))
+
+
+def send_moderator_action_email(
+        sender_id, recipient_id, intent, exploration_title, email_body):
+    """Sends a email immediately following a moderator action (publicize,
+    unpublish, delete) to the given user.
+
+    The caller is responsible for ensuring that emails are allowed to be sent
+    to users (i.e. feconf.CAN_SEND_EMAILS_TO_USERS is True).
+    """
+    require_moderator_email_prereqs_are_satisfied(intent)
+
+    recipient_user_settings = user_services.get_user_settings(recipient_id)
+    sender_user_settings = user_services.get_user_settings(sender_id)
+    email_subject = feconf.VALID_MODERATOR_ACTIONS[intent]['email_subject_fn'](
+        exploration_title)
+    full_email_content = (
```

Comment:
I added a message in the frontend to tell them not to do that, and to explain how the email message is constructed. I don't want to implement placeholders and interpolation at the moment; this will take a while and it's possible to get it wrong.